Resources/Public/JavaScript/paste-reference-drag-drop.js

```diff
@@ -0,0 +1,328 @@
+    const $pasteAction = typeof $draggableElement === 'number' || typeof $draggableElement === 'undefined';
+    let $pasteElement = null;
+    if ($draggableElement) {
+      $pasteElement = $draggableElement;
+    } else if (typeof top.itemOnClipboardUid === 'number') {
+      $pasteElement = top.itemOnClipboardUid;
+    }
+    // send an AJAX request via the AjaxDataHandler
+    let contentElementUid = $pasteAction ? $pasteElement : null;
+    if (!contentElementUid && typeof $draggableElement.data === 'function') {
+      contentElementUid = parseInt($draggableElement.data('uid') ?? 0);
+    }
+    if (contentElementUid > 0 || (DragDrop.default.newContentElementDefaultValues.CType && !$pasteAction)) {
+      let parameters = {};
+      // add the information about a possible column position change
+      const targetFound = $droppableElement.closest(DragDrop.default.contentIdentifier).data('uid');
+      // the item was moved to the top of the colPos, so the page ID is used here
+      let targetPid = 0;
+      if (typeof targetFound === 'undefined') {
+          // the actual page is needed
+          targetPid = $('.t3js-page-ce[data-page]').first().data('page');
+      } else {
+          // the negative value of the content element after where it should be moved
+          targetPid = 0 - parseInt(targetFound);
+      }
+      const $closestElementWithLanguage = $draggableElement || $droppableElement.closest('[data-language-uid]');
+      let language = $closestElementWithLanguage;
+      if (language !== parseInt($closestElementWithLanguage)) {
+          language = parseInt($closestElementWithLanguage.data('language-uid'));
+      }
+      if (language !== -1) {
+          language = parseInt($droppableElement.closest('[data-language-uid]').data('language-uid'));
+      }
+      let colPos = 0;
+      if (targetPid !== 0) {
+          colPos = newColumn;
+      }
+      parameters['cmd'] = {tt_content: {}};
+      parameters['data'] = {tt_content: {}};
+      let copyAction = (evt && evt.originalEvent && evt.originalEvent.ctrlKey || $droppableElement.hasClass('t3js-paste-copy') || evt === 'copyFromAnotherPage');
+      if (DragDrop.default.newContentElementDefaultValues.CType) {
+        parameters['data']['tt_content']['NEW234134'] = DragDrop.default.newContentElementDefaultValues;
+        parameters['data']['tt_content']['NEW234134']['pid'] = targetPid;
+        parameters['data']['tt_content']['NEW234134']['colPos'] = colPos;
+        parameters['data']['tt_content']['NEW234134']['sys_language_uid'] = language;
+
+        if (!parameters['data']['tt_content']['NEW234134']['header']) {
+          parameters['data']['tt_content']['NEW234134']['header'] = TYPO3.l10n.localize('tx_paste_reference_js.newcontentelementheader');
+        }
+
+        parameters['DDinsertNew'] = 1;
+
+        // fire the request, and show a message if it has failed
+        AjaxDataHandler.process(parameters).theb(function (result) {
```

Comment:
Should this be "then" instead of "theb"?

Reply:
It's a pity that theb is not known in PHP, but yes, you're right. Thanks!